### ISSUE TITLE
Plugin overwrites decks after manual edits in Anki

### ISSUE THREAD

**Issue by sdnnvs** [kitschpatrol/yanki-obsidian#6]
Luckily, I used an empty deck for testing. The disaster could have been huge.

Despite the plugin instructions stating that the complex deck structure of the monitored folder works, it doesn't. All subdecks are created in the vault without any hierarchy.

The procedure was as follows:

1. In Obsidian, I created the folder monitored by the plugin called "Flashcards".
2. Created the "Subdeck" folder inside the "Flashcards" folder.
3. Sync
4. In Anki, Subdeck is created as a deck and not a subdeck of Flashcard.
5. In Anki, I move Subdeck inside the Flashcard deck. Now we have Flashcard::Subdeck.
6. In Obsidian, Sync.
7. In Anki, Subdeck becomes a deck again and the Flashcard deck is deleted with all its cards!!!

In summary, we have two problems:
1. complex decks in Obsidian are not being mirrored in Anki during synchronization;
2. when we move a deck to become a subdeck, with subsequent synchronization, the subdeck turns into a deck again and the deck that supported it is deleted.

**Comment by kitschpatrol** [comment]
This is all working as designed and documented. I'm open to changing the automatic deck behavior if it feels too unexpected.

First point — and this one I'm open to reconsidering:

**Decks are created, as needed, _if_ there is more than one sub-deck to contain in a given hierarchy.**

Currently, syncing `/Flashcards` with this folder structure:

```
📁 Flashcards
└── 📁 Subdeck
   ├── 📝 Note A.md
   └── 📝 Note B.md
```

Will yield the following decks in Anki, because `/Flashcards` does not contain multiple items:

```
Subdeck
```

As soon as you add something else to `/Flashcards`, there's something differentiable to contain, and so the containing deck will be generated:

```
📁 Flashcards
├── 📁 Subdeck
│  ├── 📝 Note 1.md
│  └── 📝 Note 2.md
└── 📁 Subdeck B
   ├── 📝 Note 1.md
   └── 📝 Note 2.md
```

Yields the following decks in Anki:

```
Flashcards::Subdeck
Flashcards::Subdeck B
```

There are cases when using the stand-alone CLI version of Yanki where this makes more sense, but I agree that in Obsidian automatically pruning the hierarchy can be unexpected, and any folder selected to sync in Obsidian should create a deck, even if it contains only a single child deck.

I'll weigh changing this behavior and post to this issue thread when a change (or a decision to leave as-is) is made.

---

Second point, and this is not subject to change:

**Your markdown notes and their hierarchy are the single point of truth.**

This is why _**editing Yanki-generated cards or decks in Anki is not recommended or supported**_. You described doing exactly this in your summary when you moved the subdeck in Anki in your step 5. When you synced again, Anki saw that some of the notes it had previously created were now in the "wrong" deck according to the logic outline above, and moved them accordingly to match the hierarchy of your synced folder. There's no reason for its opinion about which decks to create should change

And for what it's worth, any cards Anki had _not_ previously created / synced were never at risk, since Yanki only acts on Anki notes with a specific namespace field and value.

**Comment by kitschpatrol** [comment]
I'm editing the issue title to make it clear that this behavior is not happening spontaneously, since it's happening _after_ changes have been made manually to generated cards in Anki, as [advised against in the documentation](https://github.com/kitschpatrol/yanki-obsidian?tab=readme-ov-file#can-i-edit-notes-in-the-anki-app).

**Comment by kitschpatrol** [comment]
I've added some documentation to the readme explaining the logic around how folders and notes are mapped to deck.

There also _was_ one data losing bug that has been identified and fixed in issue #14 , which is possibly related to what you originally reported.

If you run into anything similar in the future and would like support, please open a new issue with a minimal sample vault and steps to reproduce.
